ChattoAdditions/Source/Input/ChatInputBar.swift

```diff
@@ -250,6 +260,11 @@ extension ChatInputBar: UITextViewDelegate {
     }
 
     public func textView(_ textView: UITextView, shouldChangeTextIn nsRange: NSRange, replacementText text: String) -> Bool {
+        if self.enableReturnKeyToSend && text.range(of: "\n") != nil {
```

Comment:
More precisely, the text insertion with paragraphs will not work at all :)
I recently solved a similar problem, the code below should work as expected:

```swift
func textView(_ textView: UITextView, shouldChangeTextIn range: NSRange, replacementText text: String) -> Bool {
	let pasteboardString = UIPasteboard.general.string
	let pasteboardStringIsInserted = (pasteboardString == text)
	if (!pasteboardStringIsInserted && (text.rangeOfCharacter(from: .newlines, options: [.backwards])) != nil) {
		// implement desired behavior here
		return false
	}
	return true
}
```

@jk2K 
Perhaps you should add additional checks for array of `strings` stored in `UIPasteboard` or implement `contains(pasteboardTypes:inItemSet:)` check, but the general idea should be clear.